qtpy/utils.py

```diff
@@ -30,3 +30,17 @@ def getattr_missing_optional_dep(name, module_name, optional_names):
     if name in optional_names:
         return _wrap_missing_optional_dep_error(attr_error, **optional_names[name])
     return attr_error
+
+
+def _possibly_static_exec(cls, *args, **kwargs):
```

Comment:
I think this should be renamed to `possibly_static_exec` (just as with `getattr_missing_optional_dep`) because it's called from other modules here, so it's basically a public function.

Reply:
Does it then get exported, like `parse`? It isn't desired behavior. I can delete it at the end, though.

Reply:
Done: https://github.com/spyder-ide/qtpy/pull/422/commits/b59f7c0ba2803660df63ef158751fa0b31fefd3a

Reply:
As I commented previously in the `getattr_missing_optional_dep` case, for a library (like QtPy) used by many downstream applications, "private"/"internal" _to the package_ (i.e. QtPy, it is not intended for outside consumers of the library to use) is the much more relevant (and important) meaning here than "private"/"internal" _to the module_ (as would be the case for an application like Spyder):

> It's private to the _package_ as a whole rather than the specific module its defined in. At present, this means it is exposed as a module-level public function in any of the modules its imported in. Furthermore, its not intended to be use by anything except as an internal implementation detail of the common `__getattr__` implementation.

Therefore, like the `getattr` helper, IMO it is entirely appropriate that it be `_` private, as it is only intended for strictly internal use rather than being exposed to users.

Perhaps, though, we could just mark the `utils` module private instead, since was specifically created for such "private"/"internal" utils.

Reply:
> Perhaps, though, we could just mark the utils module private instead (i.e. _utils), since was specifically created for such "private"/"internal" utils.

Yeah, I think that's better. And then rename all functions in it to be public if they are use in other QtPy modules?

Reply:
Should I do it in the PR?

Reply:
> And then rename all functions in it to be public if they are used in other QtPy modules?

In this case there's only the two functions, the `possibly_static_exec` added in this PR and the `getattr_missing_optional_deps` just added in another recent PR in which the module was created, and they are generally imported in the same files, so it wouldn't require any function renaming, just fixing a few imports.

> Should I do it in the PR?

I would recommend another PR, to avoid the file move obscuring the additions here.

Reply:
> so it wouldn't require any function renaming, just fixing a few imports.

There are only three functions in `utils.py` (the other one being `_wrap_missing_optional_dep_error`), so I think we should rename them all in the same PR.

> I would recommend another PR, to avoid the file move obscuring the additions here.

Agreed.